File: Client/src/app/core/models/search.model.ts
export class SearchModel {
    public query: string;
    public sort: string;
    public direction: string;
    public limit: number;
    public skip: number;
    public pageIndex: number;

    public constructor(fields?: Partial<SearchModel>) {

        if (fields) {
            Object.assign(this, fields);
        }
    }
}

File: VRP.Dtos/Core/SearchModel.cs
namespace VRP.Dtos.Core {
    public class SearchModel {
        public string Query { get; set; }
        public string Sort { get; set; }
        public string Direction { get; set; }
        public int Limit { get; set; } = 10;
        public int Skip { get; set; } = 0;
        public int PageIndex { get; set; } = 0;
    }
}
